### DISCUSSION TITLE
How to rebuild esp32-camera lib?

### DISCUSSION THREAD

**Discussion by twang220** [espressif/arduino-esp32#7813] [Q&A]
Hi everyone,

I have to change the code of esp32-camera lib (tools/sdk/esp32/lib/libesp32-camera.a), because I have to use the i2c port not only for cam, but for other type sensors(there is no other gpios usable ), and I got problem, the cam can not be inited. Solution is specify the i2c address of cam instead of using SCCB_Probe(); So I need to change the source codes and rebuild the esp32-camera.a lib.

Someone could help me? Thx!

**Comment by pedrominatel** [comment]
Hi,

You need to use the lib builder: https://docs.espressif.com/projects/arduino-esp32/en/latest/lib_builder.html

See the instructions in the provided link.

**Reply by twang220** [reply]
thanks for your reply. I have another question: where can I find the source code of esp-camera with version 1.0.6? There is only 3 old releases   which start from v2.0.0

**Reply by pedrominatel** [reply]
You can find the Arduino-ESP32 1.0.6 here: https://github.com/espressif/arduino-esp32/tree/1.0.6

**Reply by twang220** [reply]
I need the esp32-camera lib source code of version 1.0.6, the link you given above is the source code of arduino-esp32(not of the libesp32-camera.a). The last source code of esp32-camera is here: https://github.com/espressif/esp32-camera, but I need the version of 1.0.6.

**Reply by me-no-dev** [reply]
Just pick a commit dated a bit before 1.0.6 was released and you will be fine. 1.0.6 is very old and unsupported (both ESP-IDF and Arduino side)
